### PR TITLE
feat: add `enable_device_summary` flag to Trainer

### LINKED ISSUE #13378: Add `enable_device_summary` flag to disable device printout
## 🚀 Feature

Add `enable_model_summary` boolean kwarg to `pl.Trainer()` to supress `_log_device_info()`'s output.

### Motivation

When calling predict within a surrogate model loop Trainer prints out the devices each time breaking apart intended tables etc or other outputs. Related to https://github.com/Lightning-AI/lightning/issues/13358 for cleaning-up/reducing stdout verbosity.
```
GPU available: False, used: False
TPU available: False, using: 0 TPU cores
IPU available: False, using: 0 IPUs
HPU available: False, using: 0 HPUs
=======================================================
n_gen |  n_eval |  n_nds  |     eps      |  indicator  
=======================================================
    1 |     322 |       3 |            - |            -
GPU available: False, used: False
TPU available: False, using: 0 TPU cores
IPU available: False, using: 0 IPUs
HPU available: False, using: 0 HPUs
    2 |    1322 |       4 |  0.625000000 |        ideal
```

### Pitch

Add `enable_model_summary` kwarg to `Trainer` that defaults to True

### Alternatives

The suggested solution is the simplest solution, any alternative would add more complexity.

### Additional context

None

______________________________________________________________________

#### If you enjoy Lightning, check out our other projects! ⚡

- [**Metrics**](https://github.com/Lightning-AI/metrics): Machine learning metrics for distributed, scalable PyTorch applications.

- [**Lite**](https://pytorch-lightning.readthedocs.io/en/latest/starter/lightning_lite.html): enables pure PyTorch users to scale their existing code on any kind of device while retaining full control over their own loops and optimization logic.

- [**Flash**](https://github.com/Lightning-AI/lightning-flash): The fastest way to get a Lightning baseline! A collection of tasks for fast prototyping, baselining, fine-tuning, and solving problems with deep learning.

- [**Bolts**](https://github.com/Lightning-AI/lightning-bolts): Pretrained SOTA Deep Learning models, callbacks, and more for research and production with PyTorch Lightning and PyTorch.

- [**Lightning Transformers**](https://github.com/Lightning-AI/lightning-transformers): Flexible interface for high-performance research using SOTA Transformers leveraging Pytorch Lightning, Transformers, and Hydra.

cc @borda @awaelchli @ananthsub @rohitgr7 @justusschock @kaushikb11

### PR DESCRIPTION
**Pull request by crawfordxx**
## What does this PR do?

Fixes #13378

Adds a new `enable_device_summary` boolean parameter to the `Trainer` that controls whether device information (GPU/TPU availability and usage) is logged during initialization. Defaults to `True` for backward compatibility.

Setting `enable_device_summary=False` suppresses the device info output, which is useful when calling `Trainer.predict()` in a loop where the repeated device summaries clutter the output:

```
GPU available: False, used: False
TPU available: False, using: 0 TPU cores
=======================================================
n_gen |  n_eval |  n_nds  |     eps      |  indicator  
=======================================================
    1 |     322 |       3 |            - |            -
GPU available: False, used: False
TPU available: False, using: 0 TPU cores
    2 |    1322 |       4 |  0.625000000 |        ideal
```

### Usage

```python
# Suppress device info when using predict in a loop
trainer = Trainer(enable_device_summary=False)
for batch in surrogate_loop:
    trainer.predict(model, dataloaders=batch_loader)
```

### Changes

- Added `enable_device_summary: bool = True` parameter to `Trainer.__init__`
- Gated the `_log_device_info()` call with the new flag
- Updated CHANGELOG and documentation

<details>
  <summary><b>Before submitting</b></summary>

- [x] Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

----
📚 Documentation preview 📚: https://pytorch-lightning--21637.org.readthedocs.build/en/21637/

### CODE REVIEW

File: src/lightning/pytorch/CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,8 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
 - Added `suggest_integrations` flag to `Trainer` to control whether optional integration suggestions (e.g., litmodels, litlogger) are shown in logs ([#21632](https://github.com/Lightning-AI/pytorch-lightning/pull/21632))
 
+- Added `enable_device_summary` flag to `Trainer` to control whether device info (GPU/TPU availability) is logged at initialization ([#13378](https://github.com/Lightning-AI/pytorch-lightning/issues/13378))
```
Comment (deependujha) [inline src/lightning/pytorch/CHANGELOG.md]:
```suggestion
- Added `enable_device_summary` flag to `Trainer` to control whether device info (GPU/TPU availability) is logged at initialization ([#21637](https://github.com/Lightning-AI/pytorch-lightning/issues/21637))
```

**Comment by deependujha** [comment]
Hi @crawfordxx, based on the discussion of the issue, a callback is expected. Please update your PR accordingly.
